app/app/users/[user_id]/page.tsx

```diff
@@ -51,17 +57,52 @@ export default function UserDetails({
 }) {
   const token = useToken();
   const userId = params.user_id;
-  const { data: user } = useUser(userId, token);
+  const { data: user, mutate } = useUser(userId, token);
+  const [showNewUserModal, setShowNewUserModal] = useState(false);
+  const onCloseModal = () => setShowNewUserModal(false);
+
+  const onUpdateUsercallback = useCallback(async () => {
+    // refetch the user details and close
+    await mutate();
```

Comment:
so `mutate` functions as a refetch?

Reply:
yes. with SWR, you can call `mutate()` to refetch, or "revalidate" in their terms. [Here](https://swr.vercel.app/docs/mutation#revalidation) are the docs.

they use term `mutate` because you can also use this function to update the in-memory data cache.